lua/barbar/ui/render.lua

```diff
@@ -607,12 +607,14 @@ local function generate_tabline(bufnrs, refocus)
       end
     end
 
-    do
-      local inactive_separator = config.options.icons.inactive.separator.left
-      if inactive_separator ~= nil and #unpinned > 0 and
-        data.buffers.unpinned_width + strwidth(inactive_separator) <= data.buffers.unpinned_allocated_width
-      then
-        content = nodes.insert(content, data.buffers.used_width, { text = inactive_separator, hl = HL.SIGN_INACTIVE })
+    if config.options.icons.separator_at_end then
+      do
```

Comment:
You can get rid of this `do`, that was only to scope the variables inside of it (which is done by the `if` now)

Reply:
Ah, true, fixed.